src/client/components/string-filter-menu/string-filter-menu.tsx

```diff
@@ -267,23 +271,26 @@ export class StringFilterMenu extends React.Component<StringFilterMenuProps, Str
       }
 
       var checkboxType = filterMode === Filter.EXCLUDED ? 'cross' : 'check';
-
-      rows = rowData.map((d) => {
-        var segmentValue = d[dimension.name];
-        var segmentValueStr = String(segmentValue);
-        var selected = selectedValues && selectedValues.contains(segmentValue);
-
-        return <div
-          className={'row' + (selected ? ' selected' : '')}
-          key={segmentValueStr}
-          title={segmentValueStr}
-          onClick={this.onValueClick.bind(this, segmentValue)}
-        >
-          <div className="row-wrapper">
-            <Checkbox type={checkboxType as CheckboxType} selected={selected}/>
-            <HighlightString className="label" text={segmentValueStr} highlightText={searchText}/>
-          </div>
-        </div>;
+      var promotedElements = promotedValues ? promotedValues.elements : [];
+
+      rows = promotedElements.concat(rowData)
```

Comment:
this is really yucky type mixing, you get an array like ["blah1", "blah2", { page: "blah3" }]

why not just do `promotedElements.concat(rowData.map((d) => d[dimension.name])` ?